lib/pure/collections/lists.nim

```diff
@@ -458,20 +458,32 @@ proc prepend*[T](L: var SinglyLinkedList[T], value: T) {.inline.} =
     assert a.contains(9)
   prepend(L, newSinglyLinkedNode(value))
 
-proc add*[T](L1: var SinglyLinkedList[T], L2: SinglyLinkedList[T]) {.since: (1, 5).} =
-  ## Adds `L2` to the end of `L1`. Efficiency: O(1).
-  ## Note that the two lists share structure after the operation.
+func copy*[T](L: SinglyLinkedList[T]): SinglyLinkedList[T] {.since: (1, 5).} =
+  ## Creates a shallow copy of `L`.
   runnableExamples:
-    var a = [1, 2, 3].toSinglyLinkedList
-    let b = [4, 5].toSinglyLinkedList
+    let a = [1, 2, 3].toSinglyLinkedList
+    assert $a == $a.copy
```

Comment:
can you expand on this example to illustrate the semantics of shallow copying? eg using a ref object as in https://github.com/nim-lang/RFCs/issues/303#issuecomment-747350870 or something better

Reply:
OK